bin/python/genomes/parse_genome_annotations.py

```diff
@@ -0,0 +1,204 @@
+from persist_annotation_metadata import *
+from utils import *
+
+parser = argparse.ArgumentParser(
+    description=__doc__, # Use docstring at top of file for --help summary
+    formatter_class=argparse.RawDescriptionHelpFormatter)
+parser.add_argument('--use_cache',
+                    help='Whether to use cache',
+                    action='store_true')
+parser.add_argument('--vault_path',
+                    help='Path in Vault for GCS service account credentials')
+parser.add_argument('--local_output_dir',
+                    help='Local directory for output.  Default: output/',
+                    default='output/')
+parser.add_argument('--gcs_bucket',
+                    help='Name of GCS bucket for upload.  ' +
+                    'Default: reference_data_dev/',
+                    default='fc-bcc55e6c-bec3-4b2e-9fb2-5e1526ddfcd2')
+parser.add_argument('--remote_output_dir',
+                    help='Remote directory for output in GCS bucket.  ' +
+                    'Default: reference_data_dev/',
+                    default='reference_data_dev/')
+parser.add_argument('--copy_data_from_prod_dir',
+                    help='Remote directory from which to copy data into ' +
+                    'remote_output_dir.  Use to ensure test data ' +
+                    'environment is equivalent to production data ' +
+                    'environment.  Default: reference_data/',
+                    default='reference_data/')
+args = parser.parse_args()
+use_cache = args.use_cache
+vault_path = args.vault_path
+gcs_bucket = args.gcs_bucket
+output_dir = args.local_output_dir
+remote_prod_dir = args.copy_data_from_prod_dir
+remote_output_dir = args.remote_output_dir
+
+scp_species = get_species_list('organisms.tsv')
+
+context = {
+    'vault_path': vault_path,
+    'gcs_bucket': gcs_bucket,
+    'output_dir': output_dir,
+    'remote_prod_dir': remote_prod_dir,
+    'remote_output_dir': remote_output_dir
+}
+
+def get_ensembl_metadata():
+    """Get organism, assembly, and annotation release metadata from Ensembl
+    """
+    ensembl_metadata = {}
+
+    # API docs: https://rest.ensembl.org/documentation/info/species
+    url = 'https://rest.ensembl.org/info/species?content-type=application/json'
+    with request.urlopen(url) as response:
+        data = response.read().decode('utf-8')
+    ensembl_species = json.loads(data)['species']
+
+    for species in ensembl_species:
+        taxid = species['taxon_id']
+        name = species['name']
+        assembly = species['assembly']
+        strain = species['strain']
+
+        if (taxid == '10090' and strain != 'reference (CL57BL6)'):
+            # Mouse has multiple annotated assemblies; only use reference assembly
+            continue
+
+        ensembl_metadata[taxid] = {
+            'organism': name,
+            'taxid': taxid,
+            'assembly_name': assembly,
+            'assembly_accession': species['accession'],
+            'release': str(species['release'])
```

Comment:
Wherever we're getting a release date, we need to check the format of that date.  YYYY-MM-DD is fine, as is YYYY/MM/DD, or similar formats.  But just YYYY-MM won't save - we have to make sure we have a day.  If the day isn't present, we should default to '01'.

Reply:
Fixed!